### ISSUE TITLE
[Bug]  XTTS v2 output corrupted with transformers>=4.52

### ISSUE THREAD

**Issue by Agatharsimao** [idiap/coqui-ai-TTS#394]
### Describe the bug

Hi,
I'm currently testing xtts_v2, and I've encountered a platform-specific issue: the same code with the same dependencies produces different audio results on Windows and Linux.

Linux: Works perfectly — audio is clear and intelligible.

Windows: The output audio consists only of distorted or "weird" noise, regardless of voice or language.

What I’ve Tried:
Executing the output WAV at different sample rates.

Using various voice and language combinations.

Fresh reinstall of all dependencies and models.

Running both with and without CUDA.

Confirmed speaker index is valid (tts.speakers[4]).

Checked audio playback with multiple tools.

Unfortunately, none of these changes resolved the issue on Windows.

`import time
import torch
from TTS.api import TTS
import sounddevice as sd
import soundfile as sf

text = "It took me quite a long time to develop a voice, and now that I have it I'm not going to be silent."

device = "cuda" if torch.cuda.is_available() else "cpu"
print(f"Device: {device}")

# Load model
tts = TTS("tts_models/multilingual/multi-dataset/xtts_v2").to(device)
print(f"Available speakers: {tts.speakers}")

start = time.time()

tts.tts_to_file(
    text=text,
    file_path="output.wav",
    speaker=tts.speakers[4],  # Make sure this is valid
    language="en",
    split_sentences=False,
)

end = time.time()
print("Time taken: {:.2f}s".format(end - start))

# Play audio
data, samplerate = sf.read("output.wav")
print(f"Sample rate: {samplerate}, Shape: {data.shape}")

sd.play(data, samplerate)
sd.wait()
`

here it is my pip --list
`Package                    Version
-------------------------- -----------
absl-py                    2.2.2
aiohappyeyeballs           2.6.1
aiohttp                    3.11.18
aiosignal                  1.3.2
anyascii                   0.3.2
attrs                      25.3.0
audioread                  3.0.1
babel                      2.17.0
certifi                    2025.4.26
cffi                       1.17.1
charset-normalizer         3.4.2
colorama                   0.4.6
contourpy                  1.3.2
coqpit-config              0.2.0
coqui-tts                  0.26.1
coqui-tts-trainer          0.2.3
cycler                     0.12.1
Cython                     3.1.1
dateparser                 1.1.8
decorator                  5.2.1
docopt                     0.6.2
einops                     0.8.1
encodec                    0.1.1
filelock                   3.18.0
fonttools                  4.58.0
frozenlist                 1.6.0
fsspec                     2025.5.0
grpcio                     1.71.0
gruut                      2.4.0
gruut-ipa                  0.13.0
gruut_lang_de              2.0.1
gruut_lang_en              2.0.1
gruut_lang_es              2.0.1
gruut_lang_fr              2.0.2
huggingface-hub            0.31.4
idna                       3.10
inflect                    7.5.0
Jinja2                     3.1.6
joblib                     1.5.0
jsonlines                  1.2.0
kiwisolver                 1.4.8
lazy_loader                0.4
librosa                    0.11.0
llvmlite                   0.44.0
Markdown                   3.8
MarkupSafe                 3.0.2
matplotlib                 3.10.3
monotonic-alignment-search 0.2.0
more-itertools             10.7.0
mpmath                     1.3.0
msgpack                    1.1.0
multidict                  6.4.4
networkx                   3.4.2
num2words                  0.5.14
numba                      0.61.2
numpy                      2.2.6
packaging                  25.0
pillow                     11.2.1
pip                        25.0.1
platformdirs               4.3.8
pooch                      1.8.2
propcache                  0.3.1
protobuf                   6.31.0
psutil                     7.0.0
pycparser                  2.22
pyparsing                  3.2.3
pysbd                      0.3.4
python-crfsuite            0.9.11
python-dateutil            2.9.0.post0
pytz                       2025.2
PyYAML                     6.0.2
regex                      2024.11.6
requests                   2.32.3
safetensors                0.5.3
scikit-learn               1.6.1
scipy                      1.15.3
setuptools                 80.8.0
six                        1.17.0
sounddevice                0.5.2
soundfile                  0.13.1
soxr                       0.5.0.post1
sympy                      1.14.0
tensorboard                2.19.0
tensorboard-data-server    0.7.2
threadpoolctl              3.6.0
tokenizers                 0.21.1
torch                      2.7.0
torchaudio                 2.7.0
tqdm                       4.67.1
transformers               4.52.2
typeguard                  4.4.2
typing_extensions          4.13.2
tzdata                     2025.2
tzlocal                    5.3.1
urllib3                    2.4.0
Werkzeug                   3.1.3
yarl                       1.20.0`

here is the audio:  
https://github.com/user-attachments/assets/9b1dfbc8-c98c-45f6-bbeb-41869e789629
(i did a conversion to mp4 so i can upload it to github)

Let me know if there's anything else I can try to help troubleshoot this further.

Thanks!

### To Reproduce

fresh install the coqui-tts and the dependencies of this test code, run the code and listen to the audio

### Logs

```shell

```

### Environment

```shell
{
    "CUDA": {
        "GPU": [],
        "available": false,
        "version": null
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "2.7.0+cpu",
        "TTS": "0.26.1",
        "numpy": "2.2.6"
    },
    "System": {
        "OS": "Windows",
        "architecture": [
            "64bit",
            "WindowsPE"
        ],
        "processor": "AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD",
        "python": "3.12.0",
        "version": "10.0.26100"
    }
}
```

### Additional context

on the enviroment is showing up no cuda available because i did a fresh install without the cuda pytorch. however, on my real project i am having the same issue with cuda

**Comment by coffee-dark** [comment]
Hi, I have exact same issue on linux (ubuntu 22.04) with a fresh install. Synthesis works fine with model "tts_models/fr/css10/vits" but xtts models (tts_models/multilingual/multi-dataset/xtts_v2 or tts_models/multilingual/multi-dataset/xtts_v1.1) generates similar strange voices output files.

**Comment by jf87** [comment]
Downgrading these dependencies seems to fix it for me:

```
uv pip install torch==2.1.0 torchaudio==2.1.0 transformers==4.30.2 numpy==1.26.4
```

**Comment by jf87** [comment]
You can define this as requirements.txt and it should work (newer versions as posted before):

```
transformers==4.51.3
torch==2.6
torchaudio==2.6
numpy==2.2.2
coqui-tts
```

**Comment by coffee-dark** [comment]
@jf87 Thanks for your comment, works like a charm now !

**Comment by Agatharsimao** [comment]
yup! it works!!! thx a lot @jf87 and @coffee-dark  :)

**Comment by eginhard** [comment]
Thank you for testing different package versions! The issue seems to be from the transformers 4.52 release. I'll prepare a new Coqui release restricting that version until we know how to make it compatible (help with that is welcome if somebody wants to look into it).

**Comment by coffee-dark** [comment]
Well, thank you guys for maintaining this project alive, results are outstanding.

**Comment by ROBERT-MCDOWELL** [comment]
seems that transformers 4.52.2 is just out and maybe they fixed it since there is 3 versions in 2 days.
https://libraries.io/pypi/transformers/4.52.2

**Comment by eginhard** [comment]
I just released coqui-tts 0.26.2 that should prevent this issue by restricting the installation to transformers<4.52.

> seems that transformers 4.52.2 is just out and maybe they fixed it since there is 3 versions in 2 days.
> https://libraries.io/pypi/transformers/4.52.2

No, the issue occurs with 4.52.2. It's probably not a bug in transformers, but an API change that affects XTTS. I opened #400 to track looking into making it compatible again if anyone is interested in debugging.
